src/stream/src/executor/sync_kv_log_store.rs

```diff
@@ -430,6 +451,9 @@ impl<S: StateStoreRead> ReadFuture<S> {
                 *self = ReadFuture::Idle;
             }
             ReadFuture::ReadingFlushedChunk { .. } | ReadFuture::Idle => {}
+            ReadFuture::Paused => {
+                unreachable!("should not be polled after paused")
```

Comment:
Can just move the `return pending().await` at the beginning to here to avoid an extra match.